extractors/tiffimagingextractor/tiffimagingextractor.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+from pathlib import Path
+from ..imagingextractor import ImagingExtractor
+from ..extraction_tools import ArrayType, PathType
+
+
+class TiffImagingExtractor(ImagingExtractor):
+    def __init__(self, filepath: PathType, sampling_frequency=None,
+                 channel_names=None):
+
+        ImagingExtractor.__init__(self)
+        self.filepath = Path(filepath)
+        self._sampling_frequency = sampling_frequency
+        assert self.filepath.suffix in ['.tiff', '.tif']
+        # TODO placeholder
+        self._video = np.load(self.filepath)
+        self._channel_names = channel_names
+
+        if len(self._video.shape) == 3:
+            # 1 channel
+            self._num_channels = 1
+            self._num_frames, self._size_x, self._size_y = self._video.shape
```

Comment:
it seems like this will come up many times. Let's think about refactoring it into a parent class or mixin